src/app/projects/api/route.ts

```diff
@@ -8,3 +10,49 @@ export async function GET() {
 
   return NextResponse.json({ projects });
 }
+
+export async function POST(req: NextRequest) {
+
+  const { name, description, status, base64data, file_type} = await req.json();
+
+  const client = new S3Client({
+    region: process.env.AWS_REGION as string,
+    credentials: {
+      accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+      secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+    },
+  });
+
+  const prismaCreationRequest: ProjectCreationRequest = {
+    name: name,
+    description: description,
+    status: status,
+    logo: ""
+  };
+
+  const projectService = new ProjectService();
+  const res = await projectService.postProject(prismaCreationRequest)
+    .then(async (res) => {
+      console.log(`Created project with id: ${res.id}`)
+
+      const putObjectParams = {
+        Bucket: process.env.BUCKET_NAME,
+        Key: res.id.toString(),
+        Body: base64data, //TODO: Replace with multipart uplaod
+        ContentType: file_type 
+      };
+
+      return await client.send(new PutObjectCommand(putObjectParams))
+        .then(async (resAWS) => {
+          await projectService.putProjectImage(res.id, res.id.toString())
+          console.log(`Uploaded logo to S3 with id: ${res.id}`);
+        }
+        )
+        .catch((error) => {
+          console.log("Failed to Upload Image");
+        });
+    })
+    .catch((error) => { console.log(error); });
+
+  return NextResponse.json({ res });
```

Comment:
Be intentional at names, especially at boundary points. Calling `NextResponse.json({ res })` is going to result in a JSON response with res being the only key.

IE
```json
{
  "res" : {
     "name": "Cool project",
      ...
   }
}
```

While this isn't the worst, it is important to pay attention to contract points like this. APIs often have many consumers and it is important that any keys returned by them are easily understood.

I would instead simply return an object containing a project `NextResponse.json(res)` so that the project object stored at res is returned, rather than an adding an external wrapping object.